src/main/java/org/mockito/Mockito.java

```diff
@@ -1772,6 +1794,38 @@ public static MockingDetails mockingDetails(Object toInspect) {
     }
 
     /**
+     * Downcasts mock object to a specific type. This is helpful for Java 8 where
+     * <code>mock(Predicate.class)</code> cannot provide a compiler-friendly value
+     * for a variable of, say, <code>Predicate&lt;String&gt;</code>. If you
+     * wrap the call to <code>mock</code> in <code>typed(...)</code>, then you will not
+     * get compiler warnings.
+     * <p>
+     *
+     * <pre class="code"><code class="java">
+     *   Foo&lt;Bar&gt; mock = typed(mock(Foo.class));
+     * </code></pre>
+     *
+     * <p>
+     * <em>Warning:</em> this method could be accidentally used to downcast to
+     * a subclass of the mock, resulting in a <code>ClassCastException</code>.
+     * For example you cannot write:
+     * <p>
+     *
+     * <pre class="code"><code class="java">
+     *     // the mock is NOT an array list, so this will throw an exception
+     *     ArrayList&lt;String&gt; list = typed(mock(List.class));
+     * </code></pre>
+     *
+     * @param mock the return of a call to one of the <code>mock</code> methods
+     * @param <T> the return type (inferred by Java 8)
+     * @param <M> the actual type of the mock
+     * @return a downcasted version of the mock
+     */
+     public static <M, T extends M> T typed(M mock) {
```

Comment:
I wonder if the same can be applied directly on `Mockito.mock(..)`?

Reply:
Note: The type variable M can be replaced by Object. The current signature is equal to `<T> T typed(Object mock)`.
At the end of the day this PR safes a compile time warning in exchange of compiler error in case nonsens is passed.

Reply:
I couldn't make the above work directly on the `mock` method - I suspect it's an issue with Java 7's inferred type system, but some of the tests stopped compiling.